woocommerce/woocommerce
# Add customize store colors

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #39716.

Implement customize store color palette feature.

Figma: Mrk6SERPZ4KrFHSjM0a8TK-fi-2631%3A258874


https://github.com/woocommerce/woocommerce/assets/4344253/1f987b2a-818d-427b-a9cb-a3d4785f0ff0




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Use WP 6.3 
2. Install the `WooCommerce Beta Tester` plugin
3. Go to /wp-admin/tools.php?page=woocommerce-admin-test-helper and enable `customize-store` feature flag
4. Use a block theme with a logo block, such as `FSE Empire` theme, or add a logo block to your theme using site-editor.
5. Go to `WooCommerce > Home`
6. Click on `Customize your store` task 
7. Click on `Assembler Hub` button
8. Click on "Change the color palette"
9. Confirm the UI matches the design
10. Select a color
11. Observe that the preview frame style should be changed.
12. Click on "or create your own"
13. Change "Background" color
14. Observe that the preview frame background color should be changed.
15. Click on "save" button
16. Changes should persist when reloading the page
17. Select a color without saving
18. Go back to main sidebar
19. Observe that changes should be discarded.





### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Add customize store color palettes

#### Comment 

</details>
